c2corg_api/tests/views/test_document_delete.py

```diff
@@ -0,0 +1,562 @@
+            ]
+        )
+        self.session.add(self.article1)
+        self.session.flush()
+
+        DocumentRest.create_new_version(self.article1, user_id)
+        update_feed_document_create(self.article1, user_id)
+        self._add_association(self.route2, self.article1)
+        self._add_association(self.outing2, self.article1)
+        self.session.flush()
+
+        self.book1 = Book(
+            activities=['skitouring'], book_types=['biography'],
+            locales=[
+                DocumentLocale(
+                    lang='en', title='Some book',
+                    description='Some content',
+                    document_topic=DocumentTopic(topic_id=9))
+            ]
+        )
+        self.session.add(self.book1)
+        self.session.flush()
+
+        DocumentRest.create_new_version(self.book1, user_id)
+        update_feed_document_create(self.book1, user_id)
+        self._add_association(self.book1, self.route2)
+        self._add_association(self.book1, self.route3)
+        self.session.flush()
+
+        self.xreport1 = Xreport(
+            activities=['hiking'], event_type=['stone_fall'],
+            locales=[
+                XreportLocale(
+                    lang='en', title='Lac d\'Annecy',
+                    place='some place descrip. in english',
+                    document_topic=DocumentTopic(topic_id=10)),
+                XreportLocale(
+                    lang='fr', title='Lac d\'Annecy',
+                    place='some place descrip. in french',
+                    document_topic=DocumentTopic(topic_id=11))
+            ]
+        )
+        self.session.add(self.xreport1)
+        self.session.flush()
+
+        DocumentRest.create_new_version(self.xreport1, user_id)
+        update_feed_document_create(self.xreport1, user_id)
+        self._add_association(self.outing2, self.xreport1)
+        self._add_association(self.route3, self.xreport1)
+        self.session.flush()
+
+        self.image1 = Image(
+            filename='image1.jpg',
+            activities=['paragliding'], height=1500,
+            image_type='collaborative',
+            locales=[
+                DocumentLocale(
+                    lang='en', title='Mont Blanc from the air',
+                    description='...',
+                    document_topic=DocumentTopic(topic_id=12))])
+        self.session.add(self.image1)
+        self.session.flush()
+
+        DocumentRest.create_new_version(self.image1, user_id)
+        self._add_association(self.outing1, self.image1)
+        self._add_association(self.route3, self.image1)
+        self._add_association(self.waypoint3, self.image1)
+        self.session.flush()
+
+        self.image1.filename = 'image1.1.jpg'
+        self.session.flush()
+        DocumentRest.update_version(
+            self.image1, user_id,
+            'changed filename', [UpdateType.FIGURES], [])
+        self.session.flush()
+
+        # TODO test deleting SVG images?
```

Comment:
Handling this should be in the responsibility of the image service: https://github.com/c2corg/v6_images/blob/master/tests/inside/test_views.py#L27